### ISSUE TITLE
Why relive vr looks horrible?

### ISSUE THREAD

**Issue by Falita7** [GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR#121]
Oh boy did i try, but can't make relive look any decent compared to Airlink (h264) 90hz 1.5x or VirtualDesktop (h265) Ultra preset. Using the relive beta 2.0 and latest driver 22.1.2 and also enabled the web access to relive, messing with render scale or encode doesn't do anything, steamvr looks the same, low res.

![000](https://user-images.githubusercontent.com/93270006/150897218-a9288b08-581b-4b18-8355-bb01c839f58e.PNG)
![com amd wirelesshmd-20220124-221007](https://user-images.githubusercontent.com/93270006/150897220-5751aa06-cbf9-4334-913e-ca1068da7b4c.jpg)
![VirtualDesktop Android-20220124-225722](https://user-images.githubusercontent.com/93270006/150897222-a7dc7851-64c0-4234-b9cf-c95c67c146f8.jpg)

What's the secret?, enabling h265 and 170mb and still the same ugly res, changed from the web server the render to 2688x2784 which is the same virtual desktop uses and still looks terrible.

**Comment by Falita7** [comment]
The clear and detailed picture comes from VD.

**Comment by GennadiyAMD** [comment]
@Falita7 - could you attach the c:\Program Files (x86)\Steam\logs\AMDWirelessVR.log and your settings.json?

**Comment by Falita7** [comment]
> @Falita7 - could you attach the c:\Program Files (x86)\Steam\logs\AMDWirelessVR.log and your settings.json?

Sure, I restored everything by default.
[AMDWirelessVR.log](https://github.com/GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR/files/7930554/AMDWirelessVR.log)

Interesting than on pictures looks a bit blurry only, but inside the headset is really blurry and with jagged edges, like scaling from 480p to 1440p levels of terrible.
![com amd wirelesshmd-20220125-010813](https://user-images.githubusercontent.com/93270006/150909685-1c023d15-c77e-4769-9368-96745c56765a.jpg)

**Comment by rbray89** [comment]
You can update the eye resolution to be closer to that of the headset. The oculus quest 2 is 1832 × 1920.

**Comment by GennadiyAMD** [comment]
@Falita7 - is it the colors that is a problem or the resolution? To me the main difference between these images is color - ReLive VR capture seems to be more reddish?

Regarding the resolution, which GPU do you have? From the log you sent I see that you're streaming at 1440x1440 per eye. On anything Navi (RX5x00 and RX6x00) you can go higher, say 1700x1700 per eye. But I'm talking about Encode resolution, not Render resolution. Render resolution is mostly a hint to the game - not all of them respect it. Then whatever is rendered is scaled to the encoder resolution, and if that's 1440x1440, it doesn't really matter what resolution the image is rendered at.

My suggestion: set encode resolution to about 1600x1600 or higher unless you're on RX470/480/580/590 and set scaling to Non-linear, which would give you more resolution in the center of the frame where the vision is the sharpest and the lens distortion is the lowest.

**Comment by Falita7** [comment]
> 

I have a 6900XT, the problem is resolution, on pictures it doesn't show how bad it really looks, is like upscaled 480p, the picture from VirtualDesktop have a color vibrance applied that's why is a little different than airlink, but also relive is more reddish.

Using this:
![Captura](https://user-images.githubusercontent.com/93270006/151088777-27d721ec-2b90-4b70-b1d6-6c91df30bc24.PNG)

Looks better but no way near how the other apps look, still blurry, bad definition on objects. Using 2000x2000 on render and encode i get a black screen.

**Comment by GManzke** [comment]
I'm having the same problem, seems like on ReLive textures are washed out compared to AirLink

**Comment by Falita7** [comment]
> I'm having the same problem, seems like on ReLive textures are washed out compared to AirLink

Yeah, I searched a lot but seems like no one got this results or maybe they thought this is how Vr should look, which is terrible. For now Airlink works ok but when you compare the visual quality that HEVC encoding offers on Virtual Desktop is hard to go back, specially on games were you have many shadows or dark areas, the banding and artifacts of compression are noticeable on h264 (Airlink) compared to VD (h265).

**Comment by GennadiyAMD** [comment]
Zoomed in - this would be controlled by the FOV setting - it is stored in radians in settings.json and converted to degrees in the Web UI. Try increasing it to perhaps 104 degrees. If the textures are rendered at a certain resolution, but you zoom in because of the narrower FOV, maybe that's what's blurring them.

Another possibility - Eye resolution is just a hint to Steam and the game. It may or may not honor it based on some criteria that we have no control over and no knowledge of. There are settings in the SteamVR menu (Settings->Video->Render Resolution - change to Custom and try to specify something closer to and above your desired encode resolution).

The reddish tint - is it only noticeable in one game, or across multiple titles? What about SteamVR home - do colors look different to you there when you compare ReLive VR to VD or Airlink?

**Comment by Falita7** [comment]
Zoomed in - increasing Fov to 120 seems to fix the problem, haven't compared with AirL or VD but is close.
Reddish tint - the pictures were from the steamvr home (alyx theme), inside the game it does has a red tint, the hands on my character in Hellspit Arena are very red so is more noticeable there.

Changing the render res above 1700x1700 on relive doesn't seem to do anything (above 2000x2000 is black screen), also changing the res on steam vr doesn't seem to do anything, opened VD then opened steam vr to see the settings and the res is the same i chose on the program (Ultra preset) 2688x2784 but using this value on steam vr with relive doesn't seem to do anything.

Also is using 72hz when in-game and i don't see where to change it. The others apps let you choose fps/hz target and quality with a res slider in the case of AirL and from a list of presets on VD, no need to mess with settings on steamvr.

The streaming part works perfectly tho.

**Comment by knightsun2010** [comment]
![setting](https://user-images.githubusercontent.com/8508124/158096247-f5ca5e02-6350-4653-a263-501344092635.png)
@Falita7 
Try this setting and see if there is any improvement. the decoder resolution looks like was limited to 1920*1920, any value higher than it causes a black screen, low frame rate provides better image quality, try 72Hz.

**Comment by pendragonmikel** [comment]
so I tried using your settings, but I'm seeing that even though the video bitrate is set to 170Mbps, it really only runs at 100Mbps. Are you seeing this too?

**Comment by knightsun2010** [comment]
@pendragonmikel I guess maybe you are using 1.0.26 apk? try v2.0 apk with latest driver. it runs stable at around 150Mbps if I set the bitrate to 170Mbps.

**Comment by pendragonmikel** [comment]
Thank you @knightsun2010. That was it. I didn't realize that I had to manually sideload the apk. I had installed through sidequest originally.

**Comment by rbray89** [comment]
> 

Where can the v2 apk be found?

**Comment by GennadiyAMD** [comment]
@rbray89 - [here](https://github.com/GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR/releases). v2.0 is still in beta, that's why we haven't pushed it to Sidequest yet.

**Comment by GennadiyAMD** [comment]
@knightsun2010 - thank you very much for answering this and helping the community!

**Comment by knightsun2010** [comment]
@GennadiyAMD No problem!

**Comment by Falita7** [comment]
> ![setting](https://user-images.githubusercontent.com/8508124/158096247-f5ca5e02-6350-4653-a263-501344092635.png) @Falita7 Try this setting and see if there is any improvement. the decoder resolution looks like was limited to 1920*1920, any value higher than it causes a black screen, low frame rate provides better image quality, try 72Hz.

Tried the exact same settings, first launch i saw the steam home in a red wireframe mode with black background for some crazy reason, then it showed fine but still blurry compared to airlink or virtualdesktop, messing with fov doesn't help much and after 104 it starts to look worst. 

Also after driver 21.10.2 any newer will crash virtual desktop when launching a game.

**Comment by rbray89** [comment]
> > ![setting](https://user-images.githubusercontent.com/8508124/158096247-f5ca5e02-6350-4653-a263-501344092635.png) @Falita7 Try this setting and see if there is any improvement. the decoder resolution looks like was limited to 1920*1920, any value higher than it causes a black screen, low frame rate provides better image quality, try 72Hz.
> 
> Tried the exact same settings, first launch i saw the steam home in a red wireframe mode with black background for some crazy reason, then it showed fine but still blurry compared to airlink or virtualdesktop, messing with fov doesn't help much and after 104 it starts to look worst.
> 
> Also after driver 21.10.2 any newer will crash virtual desktop when launching a game.

Use UDP (lower overhead)

Disable non-linear encoding if you use full display encoding. I also tweaked FOV to better match my headset too.

**Comment by Falita7** [comment]
> > > ![setting]@Falita7 Try this setting and see if there is any improvement. the decoder resolution looks like was limited to 1920*1920, any value higher than it causes a black screen, low frame rate provides better image quality, try 72Hz.
> > 
> > 
> > Tried the exact same settings, first launch i saw the steam home in a red wireframe mode with black background for some crazy reason, then it showed fine but still blurry compared to airlink or virtualdesktop, messing with fov doesn't help much and after 104 it starts to look worst.
> > Also after driver 21.10.2 any newer will crash virtual desktop when launching a game.
> 
> Use UDP (lower overhead)
> 
> Disable non-linear encoding if you use full display encoding. I also tweaked FOV to better match my headset too.

Yeah tried that too with other variables as well, i mean if you never tried VD or Airlink then relive could offer a good experience but knowing how the others look in comparison and the things you have to tweak to make it look decent then relive is still a bad choice to use, at least the performance is great.

**Comment by rbray89** [comment]
> > > > ![setting]@Falita7 Try this setting and see if there is any improvement. the decoder resolution looks like was limited to 1920*1920, any value higher than it causes a black screen, low frame rate provides better image quality, try 72Hz.
> > > 
> > > 
> > > Tried the exact same settings, first launch i saw the steam home in a red wireframe mode with black background for some crazy reason, then it showed fine but still blurry compared to airlink or virtualdesktop, messing with fov doesn't help much and after 104 it starts to look worst.
> > > Also after driver 21.10.2 any newer will crash virtual desktop when launching a game.
> > 
> > 
> > Use UDP (lower overhead)
> > Disable non-linear encoding if you use full display encoding. I also tweaked FOV to better match my headset too.
> 
> Yeah tried that too with other variables as well, i mean if you never tried VD or Airlink then relive could offer a good experience but knowing how the others look in comparison and the things you have to tweak to make it look decent then relive is still a bad choice to use, at least the performance is great.

Relive looks better then Both VD and Airlink and has better latency. VD also crashes steam VR randomly for me.

**Comment by Falita7** [comment]
> > > > > 
> > Yeah tried that too with other variables as well, i mean if you never tried VD or Airlink then relive could offer a good experience but knowing how the others look in comparison and the things you have to tweak to make it look decent then relive is still a bad choice to use, at least the performance is great.
> 
> Relive looks better then Both VD and Airlink and has better latency. VD also crashes steam VR randomly for me.

Any driver newer than 21.10.2 will crash VD a few seconds after opening a game.

**Comment by rbray89** [comment]
> > > > > >
> > > 
> > > 
> > > Yeah tried that too with other variables as well, i mean if you never tried VD or Airlink then relive could offer a good experience but knowing how the others look in comparison and the things you have to tweak to make it look decent then relive is still a bad choice to use, at least the performance is great.
> > 
> > 
> > Relive looks better then Both VD and Airlink and has better latency. VD also crashes steam VR randomly for me.
> 
> Any driver newer than 21.10.2 will crash VD a few seconds after opening a game.

That's not the issue I see. It is a crash that usually occurs by about an hour, but can last as long as two hours, or happen after a few minutes.

**Comment by knightsun2010** [comment]
@Falita7 I bought VD as soon as I bought quest2 in 2020, VD performed not good at first but became better and better since they continue to optimize and update. VD stooped working for me from some time in q4 last year and just started working again a few days ago after the recent update(I am still using the 22.1.1 driver, so VD looks like has some problem too but they fixed it now), VD does perform better than relive now, more clearer and the frame rate is much stable than before.

**Comment by 100Mylesperhr** [comment]
How do i change Codecs and render resolution for Relive VR

**Comment by GennadiyAMD** [comment]
@100Mylesperhr - either by editing settings.json or through the [Web UI](https://github.com/GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR/wiki/Advanced-Configuration-Using-a-Web-Browser)

**Comment by Masano91** [comment]
for me  airlink and VD both are way better in terms of image and performance (6900xt 5800x3d).
